utils/lib/pb_type_xml.py

```diff
@@ -90,19 +97,180 @@ def add_switchblock_locations(xml):
     })
 
 
-def start_pb_type(
+def start_sub_tile(sub_tile_name, input_wires, output_wires):
+    """This function returns the sub tile definition for a given tile.
+
+    The returned sub tile contains only input, output and clock ports.
+    The rest of the tags are going to be added by the caller.
+    """
+    sub_tile_xml = ET.Element(
+        'sub_tile', {
+            'name': add_vpr_tile_prefix(sub_tile_name),
+            'capacity': "1",
+        }
+    )
+
+    # Input definitions for the TILE
+    sub_tile_xml.append(ET.Comment(" Sub Tile Inputs "))
+    for name in sorted(input_wires):
+        input_type = 'input'
+
+        if name.startswith('CLK_BUFG_'):
+            if name.endswith('I0') or name.endswith('I1'):
+                input_type = 'clock'
+        elif 'CLK' in name:
+            input_type = 'clock'
+
+        ET.SubElement(
+            sub_tile_xml,
+            input_type,
+            {
+                'name': name,
+                'num_pins': '1'
+            },
+        )
+
+    # Output definitions for the TILE
+    sub_tile_xml.append(ET.Comment(" Sub Tile Outputs "))
+    for name in sorted(output_wires):
+        ET.SubElement(
+            sub_tile_xml,
+            'output',
+            {
+                'name': name,
+                'num_pins': '1'
+            },
+        )
+
+    return sub_tile_xml
+
+
+def start_tile(
         tile_name,
         pin_assignments,
+        input_wires=None,
+        output_wires=None,
+        sites=None,
+):
+    """ Returns a new tile xml definition.
+
+    Input parameters are:
+        - tile_name: name to assign to the tile
+        - pin_assignments: location of the pins to correctly build pin locations
+        - input and output wires: needed to correctly assign input, output and clock ports
+        - sites: set of pb_types that are going to be related to this tile
+
+    Input/Output wires and sites are mutually exclusive.
+    Both of the parameters cannot be assigned
+
+    When sites is not None, it will contain a set of different kinds of sub tiles
```

Comment:
Does it make sense to just make two functions?

Reply:
Yeah, probably having them split is better. I have added a new `start_heterogeneous_tile` func.